content/en/blog/2022/apisix.md

```diff
@@ -0,0 +1,306 @@
+
+![Architecture-Present](/img/apisix/architecture-present.png)
+
+## Plugin Introduction
+
+The `opentelemetry` plugin of Apache APISIX implements Tracing data collection based on OpenTelemetry native standard (OTLP/HTTP), and sends it to OpenTelemetry Collector through HTTP protocol. This feature will be supported online in Apache APISIX 2.13.0.
+
+Since the Agent/SDK of OpenTelemetry has nothing to do with the back-end implementation, when the application integrates the Agent/SDK of OpenTelemetry, the user can easily and freely change the observability backend service without any perception on the application side, such as switching from Zipkin into Jaeger.
+
+The `opentelemetry` plug-in integrates the OpenTelemetry Agent/SDK in Apache APISIX, which can collect traced requests, generate `trace` and forward them to the OpenTelemetry Collector.
+
+The opentelemetry plugin is located on the Agent side in the above figure, but currently only supports the `trace` part, and does not support the `logs` and `metrics` protocols of OpenTelemetry.
+
+## Enable the Plugin
+
+You need to enable `opentelemetry` plugin and modify collector configuration in `conf/config.yaml` configuration file.
+
+We assume that you have already deployed the OpenTelemetry Collector and enabled the [OTLP HTTP Receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md).
+
+> If you have not completed the deployment, you can refer to the Scenario Example section in the next section to complete the deployment of OpenTelemetry Collector.
+
+The default port of the OTLP HTTP Receiver is `4318`, and the address of the `collector` is the HTTP Receiver address of the OpenTelemetry Collector. For related fields, please refer to the [Apache APISIX official documentation](https://apisix.apache.org/docs/apisix/next/plugins/opentelemetry/).
+
+For specific configuration, please refer to the following example:
+
+```shell
+plugins
+  ... # Other plugins that have been enabled
+  - opentelemetry
+plugin_attr:
+  ...
+  opentelemetry:
+    trace_id_source: x-request-id
+    resource:
+      service.name: APISIX
+    collector:
+      address: 127.0.0.1:4318 # OTLP HTTP Receiver address
+      request_timeout: 3
+```
+
+### Method 1: Enable the Plugin for a Specific Route
+
+In order to show the test effect more conveniently, `sampler` is temporarily set to full sampling in the example to ensure that `trace` data is generated after each request is traced, so that you can view `trace` related data on the Web UI. You can also set relevant parameters according to the actual situation.
```

Comment:
can you add a note on how sampling affects the overhead introduced by the plugin. Any perf data you can refer to would be useful in this blog post

Reply:
Yeah unfortunately we don't have that type of data. Can we skip it?

Reply:
sure, it may be a quesiton customers will have, but if we don't have answer yet, it is fine. Perhaps can be mentioned as a future work